microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/HybridDocumentationTest.java

```diff
@@ -0,0 +1,240 @@
+ * once proper customization is applied.
+ */
+@RunWith(Arquillian.class)
+@ServerSetup({ HybridDocumentationTest.OpenApiExtensionSetup.class })
+@RunAsClient
+public class HybridDocumentationTest {
+
+    private final static String PROVIDER_DEPLOYMENT_NAME = "serviceProviderDeployment";
+    private final static String ROUTER_DEPLOYMENT_NAME = "localServicesRouterDeployment";
+
+    private static ArquillianContainerProperties arquillianContainerProperties = new ArquillianContainerProperties(
+            ArquillianDescriptorWrapper.getArquillianDescriptor());
+    private static OnlineManagementClient onlineManagementClient;
+
+    @Deployment(name = PROVIDER_DEPLOYMENT_NAME, order = 1, testable = false)
+    public static Archive<?> serviceProviderDeployment() {
+        //  MP OpenAPI disabled on this Services Provider deployment for testing purposes, we don't want it here
+        String mpConfigProperties = "mp.openapi.extensions.enabled=false";
+        WebArchive deployment = ShrinkWrap.create(
+                WebArchive.class, PROVIDER_DEPLOYMENT_NAME + ".war")
+                .addClasses(ProviderApplication.class)
+                .addClasses(
+                        District.class,
+                        DistrictEntity.class,
+                        DistrictService.class,
+                        InMemoryDistrictService.class,
+                        DistrictsResource.class,
+                        RoutingServiceConstants.class)
+                .addAsManifestResource(new StringAsset(mpConfigProperties), "microprofile-config.properties")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return deployment;
+    }
+
+    @Deployment(name = ROUTER_DEPLOYMENT_NAME, order = 2, testable = false)
+    public static Archive<?> localServicesRouterDeployment()
+            throws ConfigurationException, IOException, ManagementClientRelatedException {
+
+        int configuredHTTPPort;
+        try (OnlineManagementClient client = ManagementClientProvider.onlineStandalone()) {
+            configuredHTTPPort = OpenApiServerConfiguration.getHTTPPort(client);
+        }
+
+        //  OpenAPI doc here is exposing a document generated from static file (META-INF/openapi.yaml) plus annotations
+        //  from Local Service Provider "non-routed" JAX-RS endpoints, edited through and OASModelReader implementation
+        //  and eventually filtered through a OASFilter one.
+        //  MP Config is used to tell MP OpenAPI to skip doc generation for exposed "routed" JAX-RS endpoints as service
+        //  consumers must rely to the original documentation resulting from above mentioned original static file modifications.
+        //  Here we also add config properties to reach the Services Provider.
+        String mpConfigProperties = "mp.openapi.scan.exclude.packages=org.jboss.eap.qe.microprofile.openapi.apps.routing.router.rest.routed"
```

Comment:
This is not very readable. Can you move the content to the resource file?

Reply:
I understand your point. But there's dynamic configuration involved and I wouldn't like to introduce configuration complexity for a bad worded or misplaced comment. If you agree I'd like to add this comment contents to the deployment method Javadoc. WDYT?

Reply:
Yeah, I have missed those lines at the end. Yeah a method would be better than a resource file. And what about static string and then just use `String.format`? Some memory and processor time will be saved. WDYT?

Reply:
yes, I do agree. Fixing...